src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1208,30 +1208,14 @@ private void ScanBlank()
                     cc.AddCategoryFromName(ParseProperty(), ch != 'p', UseOptionI(), _pattern, _currentPos);
                     if (UseOptionI())
                     {
-                        cc.AddLowercase(_culture);
+                        cc.AddCaseEquivalences(_culture);
                     }
 
-                    return new RegexNode(RegexNodeKind.Set, _options, cc.ToStringClass(_options));
+                    return new RegexNode(RegexNodeKind.Set, (_options & ~RegexOptions.IgnoreCase), cc.ToStringClass(_options));
```

Comment:
ToStringClass takes options because it special-cases NonBacktracking, and it special-cases NonBacktracking because of casing.  Your change correctly cleans up the body of Canonicalize, but it doesn't remove the now unused isNonBacktracking parameter, nor the RegexOptions parameter to ToStringClass.  Can you clean up all of those and the call sites?

Reply:
Will do.